CONTRIBUTING.md

```diff
@@ -0,0 +1,95 @@
+
+Style guidelines for contributing code and writing documentation can be found [here](README.md#style-guidelines).
+
+## Branching
+
+When contributing documentation to this repository, we follow a number of guidelines.
+
+### Branch Naming
+
+All branches shall have one of these names.
+
+- `#-short-name` or `short-name`: refers to a documentation update, preferably with issue number. The short name describes the topic.
+- `ci/#-short-name` or `ci/short-name`: refers to updates to the way this repository is deployed. The short name describes the feature.
+
+### Scope
+
+A branch should be **small and focused** and should be scoped to a **single specific task**. Do not combine new features and refactoring of existing code.
+
+### Pull requests and rebasing
+
+Pull requests shall close or reference issues. Please file an issue first before submitting a pull request. When submitting a pull request, please fill out all the sections in the pull request template.
+
+- **Before** a reviewer is assigned, rebasing the branch to reduce the number of commits is highly advised. We recommend self-reviewing your own pull request: making the [commit](#commit) history clean, checking for typos or incoherences, and making sure Continuous Integration passes.
+- **During** a pull request's review, do not squash commits: it makes it harder for reviewers to read the evolution of a pull request. Making the commit history denser to answer reviewers' comments is acceptable at that point.
+- Once a pull request **has been approved** by the reviewers, it can be rebased on top of its target branch before it is merged. This is an opportunity for the contributor to clean up the commit history. A reviewer can also ask specifically for a rebase.
+
+## Commits
+
+Keep the commits to be merged clean: adhere to the commit message format defined below and instead of adding and deleting files within a pull request, drop or fix the concerning commit that added the file.
+
+Interactive rebase (`git rebase -i`) can be used to edit or rewrite commits that do not follow these contribution guidelines.
+
+## <a name="commit"></a>Commit Messages
+
+The first line of a commit message is the subject. The commit message may contain a body, separated from the subject by an empty line.
+
+### Commit Subject
+
+The subject contains the concerning topic and a concise message in [the imperative mood](https://chris.beams.io/posts/git-commit/#imperative), starting with a capital. The subject may also contain references to issues or other resources.
+
+The topic is typically a few characters long and should always be present. Accepted topics are:
+
+- `doc`: documentation
```

Comment:
i also thought this at first, but it's hard habit to break if you're committing to the stack and the docs. it may also be useful to distinguish between docs and changes to the repo

Reply:
I don't think we need this